app/models/concerns/host_searching.rb

```diff
@@ -77,6 +79,8 @@ def filter_with_results_for_policy(_filter, _operator, policy_or_profile_id)
                  .or(::Profile.where(policy_id: policy_or_profile_id))
                  .or(::Profile.where(policy_id: profiles.select(:policy_id)))
 
+      Thread.current['filtered_by_profiles'] = profiles
```

Comment:
(as you mentioned) this might leak to another request.

How about making it more global like `Thread.current['scoped_search_context'] = {...}` and reset it after each call of `Model.search_for`?

```
class ApplicationRecord < ActiveRecord::Base
  def self.search_for
    super
  ensure
    # reset context
  end
```

Reply:
Good idea, or maybe we could just use https://github.com/steveklabnik/request_store ?

Reply:
Good idea, there are also some request-wide stores that use `Thread.current`, like [this](https://github.com/steveklabnik/request_store), that prevent leaking. However, I prefer your idea more :+1: